e2e_tests/provider_cfg/all/Dockerfile

```diff
@@ -55,3 +53,17 @@ RUN softhsm2-util --init-token --slot 0 --label "Parsec Tests" --pin 123456 --so
 # Add users for multitenancy tests
 RUN useradd -m parsec-client-1
 RUN useradd -m parsec-client-2
+
+USER parsec-client-1
+RUN curl https://sh.rustup.rs -sSf | bash -s -- -y
+
+USER parsec-client-2
+RUN curl https://sh.rustup.rs -sSf | bash -s -- -y
+
+# Install Rust toolchain for root
+USER root
+RUN curl https://sh.rustup.rs -sSf | bash -s -- -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+
+# Download the SPIRE server and agent
+RUN curl -s -N -L https://github.com/spiffe/spire/releases/download/v0.11.1/spire-0.11.1-linux-x86_64-glibc.tar.gz | tar xz
```

Comment:
We should really look at upstreaming these images 😬 It would save 10 min for the `all-providers` run, almost 25% of the time.